### ISSUE TITLE
Preservation of current undefined behavior of monotone constraints for multiclass

### ISSUE THREAD

**Issue by meakbiyik** [dmlc/xgboost#5554]
This will be an interesting feature request, because I am not asking xgboost team to implement something but just be aware of the behavior of their code if they are not, and possibly document it. 

When monotone constraints are used with multi:softmax objective and mlogloss, they perform in the way that forces the constrained feature -for example if the constraint is positive- to only be used to split a node if the output points to a class other than zero. Despite the fact that there is no information in any of the documentation regarding to this phenomenon, I validated this behavior by checking out the SHAP values and the source code. 

This is a very great feature, and I would like the xgboost to preserve it in the future by not having any kind of check against the objective type, and for authors to document the implications if possible. It allowed me to find a solution to a very challenging machine learning task, and I am very happy with the results :) 

Thanks in advance!

**Comment by trivialfis** [comment]
Yes.  It become quite glaring when implementing #5460 .
